third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java

```diff
@@ -418,8 +418,17 @@ private Bitmap setPixels(GifFrame currentFrame, GifFrame previousFrame) {
     // Final location of blended pixels.
     final int[] dest = mainScratch;
 
-    // clear all pixels when meet first frame
+    // clear all pixels when meet first frame and drop prev image from last loop
     if (previousFrame == null) {
+      previousImage = null;
```

Comment:
Here you probably should be releasing previousImage so the Bitmap can be re-used, see the four lines in ``clear()``: https://github.com/bumptech/glide/blob/d447e582397cc7cf3f95a9871eb18d81a917544f/third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java#L327